src/test/java/cloud/eppo/EppoClientBanditTest.java

```diff
@@ -0,0 +1,358 @@
+    subjectAttributes.put("gender_identity", "female");
+
+    BanditActions actions = new BanditActions();
+
+    BanditResult banditResult =
+        EppoClient.getInstance()
+            .getBanditAction(flagKey, subjectKey, subjectAttributes, actions, "control");
+
+    // Verify assignment
+    assertEquals("banner_bandit", banditResult.getVariation());
+    assertNull(banditResult.getAction());
+
+    // The variation assignment should have been logged
+    ArgumentCaptor<Assignment> assignmentLogCaptor = ArgumentCaptor.forClass(Assignment.class);
+    verify(mockAssignmentLogger, times(1)).logAssignment(assignmentLogCaptor.capture());
+
+    // No bandit log since no actions to consider
+    ArgumentCaptor<BanditAssignment> banditLogCaptor =
+        ArgumentCaptor.forClass(BanditAssignment.class);
+    verify(mockBanditLogger, times(0)).logBanditAssignment(banditLogCaptor.capture());
+  }
+
+  @Test
+  public void testBanditErrorGracefulModeOff() {
+    EppoClient.getInstance()
+        .setIsGracefulFailureMode(
+            false); // Should be set by @BeforeEach but repeated here for test clarity
+    try (MockedStatic<BanditEvaluator> mockedStatic = mockStatic(BanditEvaluator.class)) {
+      // Configure the mock to throw an exception
+      mockedStatic
+          .when(() -> BanditEvaluator.evaluateBandit(anyString(), anyString(), any(), any(), any()))
+          .thenThrow(new RuntimeException("Intentional Bandit Error"));
+
+      // Assert that the exception is thrown when the method is called
+      BanditActions actions = new BanditActions();
+      actions.put("nike", new Attributes());
+      actions.put("adidas", new Attributes());
+      assertThrows(
+          RuntimeException.class,
+          () ->
+              EppoClient.getInstance()
+                  .getBanditAction(
+                      "banner_bandit_flag", "subject", new Attributes(), actions, "default"));
+    }
+  }
+
+  @Test
+  public void testBanditErrorGracefulModeOn() {
+    EppoClient.getInstance().setIsGracefulFailureMode(true);
+    try (MockedStatic<BanditEvaluator> mockedStatic = mockStatic(BanditEvaluator.class)) {
+      // Configure the mock to throw an exception
+      mockedStatic
+          .when(() -> BanditEvaluator.evaluateBandit(anyString(), anyString(), any(), any(), any()))
+          .thenThrow(new RuntimeException("Intentional Bandit Error"));
+
+      // Assert that the exception is thrown when the method is called
+      BanditActions actions = new BanditActions();
+      actions.put("nike", new Attributes());
+      actions.put("adidas", new Attributes());
+      BanditResult banditResult =
+          EppoClient.getInstance()
+              .getBanditAction(
+                  "banner_bandit_flag", "subject", new Attributes(), actions, "default");
+      assertEquals("banner_bandit", banditResult.getVariation());
+      assertNull(banditResult.getAction());
+    }
+  }
+
+  @Test
+  public void testBanditLogErrorNonFatal() {
```

Comment:
@typotter test case to ensure any issues with the user-provided bandit logging method do not interfere with returning the bandit result.